dot/network/discovery.go

```diff
@@ -173,35 +189,34 @@ func (d *discovery) checkPeerCount() {
 				continue
 			}
 
-			ctx, cancel := context.WithTimeout(d.ctx, findPeersTimeout)
-			defer cancel()
-			d.findPeers(ctx)
+			d.findPeers()
 		}
 	}
 }
 
-func (d *discovery) findPeers(ctx context.Context) {
+func (d *discovery) findPeers() {
 	logger.Debug("attempting to find DHT peers...")
 	peerCh, err := d.rd.FindPeers(d.ctx, string(d.pid))
 	if err != nil {
 		logger.Warnf("failed to begin finding peers via DHT: %s", err)
 		return
 	}
 
+	timeout := time.NewTimer(findPeersTimeout)
+	defer timeout.Stop()
```

Comment:
Stop and drain the channel in the `case peer := <-peerCh:` block in case you receive the peer at the same time as the timer triggers. You can use

```go
if !timer.Stop() {
    <-timer.C
}
```